File: exercism/python/rna-transcription/rna_transcription.py
def transcribe(nucleotide: str) -> str:
    complement = {'G': 'C', 'C': 'G', 'T': 'A', 'A': 'U'}
    return complement.get(nucleotide, '')


def to_rna(dna: str) -> str:
    rna = [transcribe(n) for n in dna]
    return ''.join(rna) if '' not in rna else ''

File: exercism/typescript/rna-transcription/rna-transcription.ts
export default class Transcriptor {
    transcribe(nucleotide: string): string {
        switch (nucleotide) {
            case "G":
                return "C"
            case "C":
                return "G"
            case "T":
                return "A"
            case "A":
                return "U"
            default:
                throw new Error("Invalid input DNA.")
        }
    }

    toRna(dna: string): string {
        return dna
            .split("")
            .map(this.transcribe)
            .join("")
    }
}
